app/routes/analyzer.tsx

```diff
@@ -735,6 +751,88 @@ function AbilityPointsDetails({
   );
 }
 
+/**
+ * From an array of Main abilities, create a map of <Ability, number>, then return it as an Array after sorting by value, descending.
+ *
+ * The map describes the number of Ability chunks required to replace the main ability on a piece of gear.
+ *
+ * Extra processing is required for Main abilities that are primary slot-only abilities,
+ *  as they are comprised of 3 stackable ability chunks at a lower ability chunk count than usual.
+ */
```

Comment:
This shouldn't be a JSDOC comment since those should be used for describing to the caller what the function is. Regular comment looks good in this case.